examples/leaderregistry/main_test.go

```diff
@@ -350,6 +350,70 @@ func TestHandleLeaderTransitionGracefully(t *testing.T) {
 	}
 }
 
+// TestSurviveReplicaFailure tests high availability and fault tolerance in the event of a server failure.
+// It validates that actor invocations can still succeed despite the loss of a server replica.
+// The test spawns actors with multiple replicas, simulates a server failure by killing one of the servers,
+// and ensures that requests continue to succeed.
+//
+// Test Steps:
+// 1. Spawn actors with multiple replicas.
+// 2. Simulate a server failure by killing one of the servers.
+// 3. Validate that requests can still succeed if we set a retry policy that retries on replicas.
+// 4. Validate that all requests fail if we set a retry policy that never retries.
+//
+// This test provides insights into the system's resilience, identifies areas for optimization, and ensures comprehensive coverage
+// of server failure scenarios and request handling.
+func TestSurviveReplicaFailure(t *testing.T) {
+	lp := &leaderProvider{}
+	lp.setLeader(registry.Address{
+		IP:   net.ParseIP("127.0.0.1"),
+		Port: baseRegistryPort,
+	})
+
+	var (
+		server1, _, cleanupFn1 = newServer(t, lp, 0)
+		server2, _, cleanupFn2 = newServer(t, lp, 1)
+		server3, _, cleanupFn3 = newServer(t, lp, 2)
+	)
+	defer cleanupFn1()
+	defer cleanupFn2()
+	defer cleanupFn3()
+
+	// Sleep for a few seconds to let the servers heartbeat multiple times,
+	// allowing actor invocations to succeed.
+	time.Sleep(5 * time.Second)
+
+	options := types.CreateIfNotExist{Options: types.ActorOptions{
+		ExtraReplicas: 2,
+	}}
+	require.Eventually(t, func() bool {
+		_, err := server1.InvokeActor(
+			context.Background(), namespace, actorID(0), module, "keep-alive", nil, options)
+		require.NoError(t, err)
+		return server1.NumActivatedActors()+server2.NumActivatedActors()+server3.NumActivatedActors() == 3
```

Comment:
You should ensure that each server has ~ the same number of servers like the other tests do to make suer you're not killing a server that only has like 1 actor on it (somehow)

Reply:
Also comment why it should be 3 (1 replica of the actor on each server)

Reply:
sorry yes i meant actors (each should have 1 i guess)